### PR TITLE
Create logo and add to Readme

### LINKED ISSUE #975: Need a logo!
I was checking [nodejs logo issue](https://github.com/nodejs/node/issues/37) that have very nice ideas from the community.

Why not create a logo that identify async library? 💪

### PR DESCRIPTION
**Pull request by ivanseidel**
Issue #975 led me to a sprint to create a logo, as async is an awesome and globally used library. Now, it's readme page will feel even more alive with a few colors 😃 

* Logo file was placed into `/logo` folder. It's in `.AI` format since `.EPS` was really ruge (~5Mb)
* Readme header was created as `.jpg`. It's URL should be swapped from `ivanseidel/async` to `caolan/async` before merging.
* Icons were generated to the doc site ( #1202 )

Note: If all ok, let me know in order to change the URL in the Readme.

Fixes #975

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Async.js
+![Async Logo](https://raw.githubusercontent.com/caolan/async/master/logo/async-logo_readme.jpg)
```
Comment (megawac) [inline README.md]: can you change this to

`Async Logo`

Comment (megawac) [inline README.md]: or did you make it absolute for `npm` to work?

Comment (ivanseidel) [inline README.md]: That doesn't work... we need to access `raw.githubusercontent.com/...`, doing a relative path access, will access `github.com/caolan/logo/async-logo_readme.jpg` witch is an `404`...

Comment (ivanseidel) [inline README.md]: I made it absolute to work with NPM and also Github

Comment (ivanseidel) [inline README.md]: Pros: We could place easter eggs during christmas, and it would work everywere... ( \o/ )

Comment (megawac) [inline README.md]: http://stackoverflow.com/a/36313391/1517919
But sounds good, I'd rather gaurentee it works on npm than fiddle with npm ignore stuff

Comment (ivanseidel) [inline README.md]: That's new, sounds good both ways.. but for NPM, it isn't for sure..

**Comment by megawac** [comment]
Oh @ivanseidel can you export one more logo, without the borders?

(For the docs page)

**Comment by ivanseidel** [comment]
@megawac, should the logo exports be in `master`, or should we place it in the `site branch`?

I will export an .SVG, so that the resolution isn't a problem in the future, is that ok for you?

**Comment by gyeates** [comment]
Works for me, place it in master as just `logo.svg`

**Comment by megawac** [comment]
Works for me, place it in master as just logo.svg

**Comment by ivanseidel** [comment]
Just did. I followed the pattern for naming (`async-logo.svg`). This is good when you are trying to search something in the browser.. it gives some contextuality (if there are 321039123 `logo.svg`, you will know witch one is the one)

**Pull request merged**